### DISCUSSION TITLE
Updated TweakScale and now afraid to open saves

### DISCUSSION THREAD

**Discussion by IrFiKs** [TweakScale/TweakScale#303] [Support] [answered]
I updated TweakScale and got (figure 1) message. I don`t know how to fix this fatal errors so @Lisias I`m asking for your help.
![1](https://github.com/TweakScale/TweakScale/assets/34242362/f6866a26-b481-433b-a967-2f826f65ed20)
[KSP.log](https://github.com/TweakScale/TweakScale/files/11713536/KSP.log)

**Comment by Lisias** [comment]
Hi, @IrFiKs !

Sorry for the event, but better safe than sorry. This is rarely TweakScale's fault, however. In 99.9% of the cases it's a 3rd-party with rogue patches screwing things, and TweakScale is just yelling about the problems it finds.

I inspected your `KSP.log` and it appears to be something new, I didn't recognized the affected parts as something I had diagnosed before.

Please send me also your ModuleManager logs (you will find it on the `Logs` folder, on the same place you found `KSP.log`) and also the `ModuleManager.ConfigCache` (this one you will find on `GameData`). These files will tell me exactly what's happening on your rig and what I need to do to fix it.

Cheers!

**Reply by IrFiKs** [reply]
[ModuleManager.zip](https://github.com/TweakScale/TweakScale/files/11714367/ModuleManager.zip)
@Lisias

**Comment by Lisias** [comment]
@IrFiKs , found the problem. It's `dstt` (`DeepSky/ThorTech`). All parts from this thing is being double patched as follows:

```
                MODULE
                {
                        name = TweakScale
                        type = free_square
                        type = free_square
                }
```

This kind of double patching is tolerable nowadays, you can open your savegame (but make a backup first, just in case).

In the mean time, I'll inspect this thing closely and will give you a fix (or perhaps just add proper patching into a TweakScale Companion) hopefully at the end of this Sunday.

Let me know if anything goes south (remember: make backups of the whole `<KSP>/saves` directory before playing), but my best guest at this point is that TweakScale will cope with this mess for while (to tell you the true, it's some time since I had coped with this situation by the last time, so there's a chance that something changed on KSP and rendered TweakScale powerless to work around the problem).

I will come back to you as soon as I have the fix.

**Comment by Lisias** [comment]
Uh… wait… @IrFiKs , I think you installed ThorTech wrongly!

Assuming this is the ThorTech we are talking about:

https://forum.kerbalspaceprogram.com/topic/168326-19-thor-tech-v0968-~-aug-07-2020/

The right place to install it is on `<KSP>/GameData/DeepSky/ThorTech`, but you have it installed on `<KSP>/GameData/ThorTech`. **Too**!!

**Completely** delete `<KSP>/GameData/ThorTech` and I think you will be fine!

Let me know if this worked for you! Cheers!

For future reference:
* https://github.com/JadeOfMaar/DSCore/tree/master/GameData/DeepSky/00DeepSky
* https://github.com/JadeOfMaar/ThorTech/tree/master/GameData/DeepSky/ThorTech

**Reply by IrFiKs** [reply]
Yes, it helped thank you, but now there are new errors (fig. 1) and (fig. 2 Failed to load next plugins) @Lisias
![image](https://github.com/TweakScale/TweakScale/assets/34242362/8372545a-8a27-4fd5-8549-31e25e43b254)
![image](https://github.com/TweakScale/TweakScale/assets/34242362/09048683-a7d2-4bd7-9d0c-88193567d94d)
[KSP.log](https://github.com/TweakScale/TweakScale/files/11714441/KSP.log)

**Comment by Lisias** [comment, marked as answer]
@IrFiKs , There're some older add'ons on your rig, from a time in which KSP would load thing differently. It's more an annoyance than a problem, but it must be solved the same.

The really hairy stuff happens when a thingy inside KSP called `Assembly Loader/Resolver` breaks due some of these older practices. When this happens, TweakScale get really screwed and it will not work, and **this** will ruin your day for sure.

You need to manually remove the following DLLs:

* `D:\SteamLibrary\steamapps\common\Kerbal Space Program\GameData\MagicSmokeIndustries\Plugins\Scale_Redist.dll`
    * This is the rogue duplicate that its messing with TweakScale 
* `D:\SteamLibrary\steamapps\common\Kerbal Space Program\GameData\B9_Procedural_Structural`
    * This Add'On is broken, and it's triggering that nasty bug on the `Assembly Loader/Resolver` thingy that so screws up TweakScale royally.

You will need to reach `B9 Procedural Structural` maintainers in order to diagnose this one, as I don't know it - perhaps it's just missing a dependency, but the KSPCF's error handler is not telling me exactly what's happening (it only pinpoints the problem), so you will need to talk to the maintainers as they are the ones that know what this DLL needs.

Cheers!

**Reply by IrFiKs** [reply]
Thanks @Lisias, deleting Scale Redist.dll solved the problem. If I find out how to solve the problem with B9 Procedural Structural I will write here. Maybe it will be useful for someone.

Cheers!
